src/mlpack/methods/sparse_autoencoder/maximal_inputs.cpp

```diff
@@ -0,0 +1,83 @@
+#include "maximal_inputs.hpp"
+
+namespace mlpack {
+namespace nn {
+
+namespace {
+
+void VisualizeHiddenUnit(size_t rows, size_t cols,
+                         int squareRows,
+                         int offset,
+                         arma::mat const &input,
```

Comment:
After I merge this, do you mind if I change the ordering of the `const` to be the way the rest of the code is?  i.e. `const arma::mat&` vs. `arma::mat const &`.  They both do exactly the same thing.  If you have a specific reason for ordering it the way you have, that's fine too; I just try to make things look the same. :)